mvlearn/factorization/ajive.py

```diff
@@ -0,0 +1,937 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from scipy.sparse import issparse
+from copy import deepcopy
+from sklearn.externals.joblib import load, dump
+import pandas as pd
+from ..utils.utils import check_Xs
+from .ajive_utils.block_visualization import _data_block_heatmaps, \
+    _ajive_full_estimate_heatmaps
+
+from .ajive_utils.utils import svd_wrapper, centering
+from .ajive_utils.wedin_bound import get_wedin_samples
+from .ajive_utils.random_direction import sample_randdir
+from .ajive_utils.pca import pca
+
+
+class ajive(object):
+    r"""
+    An implementation of Angle-based Joint and Individual Variation Explained.
+    This algorithm takes multiple input views with the same number of samples
+    and decomposes them into 3 distinct matrices representing [#1ajive]_:
```

Comment:
Rephrased. All multiview algorithms require same number of samples. Pulled some details from the JIVE paper abstract, added a note that I just added for GCCA too.

An implementation of Angle-based Joint and Individual Variation Explained  [#1ajive]_.
This algorithm takes multiple views and decomposes them into 3 distinct matrices representing:
    - Low rank approximation of individual variation within each view
    - Low rank approximation of joint variation between views
    - Residual noise
An important note, AJIVE can handle any number of views, not just two.